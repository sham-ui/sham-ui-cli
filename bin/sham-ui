#!/usr/bin/env node

require( 'commander' )
    .version( require( '../package' ).version )
    .usage( '<command> [options]' )
    .command( 'new', 'generate a new project from a template' ).alias( 'n' )
    .command( 'generate', 'generate a new component' ).alias( 'g' )
    .parse( process.argv );