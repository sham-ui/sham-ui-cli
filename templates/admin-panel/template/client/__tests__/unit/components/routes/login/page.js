import { DI } from 'sham-ui';
import * as directives from 'sham-ui-directives';
import hrefto from 'sham-ui-router/href-to';
import RoutesLoginPage  from '../../../../../src/components/routes/login/page.sfc';
import renderer, { compile } from 'sham-ui-test-helpers';

afterEach( () => {
    DI.bind( 'router', null );
} );

it( 'renders correctly', () => {
    DI.bind( 'title', {
        change() {}
    } );
    DI.bind( 'router', {
        generate: jest.fn().mockReturnValueOnce( '/' ),
        activePageComponent: compile``,
        storage: {
            url: '/'
        }
    } );

    const meta = renderer( RoutesLoginPage, {
        directives: {
            ...directives,
            hrefto
        }
    } );
    expect( meta.toJSON() ).toMatchSnapshot();
} );

it( 'display errors', async() => {
    DI.bind( 'title', {
        change() {}
    } );
    DI.bind( 'router', {
        generate: jest.fn().mockReturnValueOnce( '/' ),
        activePageComponent: compile``,
        storage: {
            url: '/'
        }
    } );

    const loginMock = jest.fn();
    DI.bind( 'session', {
        login: loginMock.mockReturnValueOnce( Promise.reject( {} ) )
    } );

    const meta = renderer( RoutesLoginPage, {
        directives: {
            ...directives,
            hrefto
        }
    } );

    const formData = {
        email: 'admin@gmail.com',
        password: 'passw0rd'
    };
    const { component } = meta;
    component.container.querySelector( '[name="email"]' ).value = formData.email;
    component.container.querySelector( '[name="password"]' ).value = formData.password;
    component.container.querySelector( '[type="submit"]' ).click();

    await new Promise( resolve => setImmediate( resolve ) );

    expect( loginMock.mock.calls ).toHaveLength( 1 );
    expect( loginMock.mock.calls[ 0 ] ).toHaveLength( 2 );
    expect( meta.toJSON() ).toMatchSnapshot();
} );
