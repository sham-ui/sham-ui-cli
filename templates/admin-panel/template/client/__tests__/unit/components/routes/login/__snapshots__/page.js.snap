// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`display errors 1`] = `
Object {
  "Constructor": "RoutesLoginPage",
  "Options": Object {
    "ID": "component",
    "directives": Object {
      "EventListener": [Function],
      "disabled": [Function],
      "hrefto": [Function],
      "onclick": [Function],
      "oninput": [Function],
      "onload": [Function],
      "onresize": [Function],
      "onscroll": [Function],
      "onselect": [Function],
      "onsubmit": [Function],
      "ref": [Function],
    },
  },
  "html": "
<div class=\\"login-page\\">
  <form class=\\"form\\">
    <div class=\\"logo\\"><span><span>[</span> {{ logoText }} <span>]</span></span></div>
    <!--Logo-->
    <!--header-->
    <ul class=\\"errors-panel\\">
      <li>Login fail</li>
    </ul>
    <!--if-->
    <!--ErrorsPanel-->
    <div class=\\"form-group\\">
      <input name=\\"email\\" type=\\"text\\" placeholder=\\"Enter your email\\" autofocus=\\"\\">
    </div>
    <div class=\\"form-group\\">
      <input name=\\"password\\" type=\\"password\\" placeholder=\\"Enter your password\\">
    </div>
    <!--default-->
    <button type=\\"submit\\">Sign In
      <!--submit-->
    </button>
    <div class=\\"signup-label\\"> Not yet a member? <a href=\\"/\\"> Sign Up </a>
    </div>
    <!--footer-->
  </form>
</div>
",
}
`;

exports[`renders correctly 1`] = `
Object {
  "Constructor": "RoutesLoginPage",
  "Options": Object {
    "ID": "component",
    "directives": Object {
      "EventListener": [Function],
      "disabled": [Function],
      "hrefto": [Function],
      "onclick": [Function],
      "oninput": [Function],
      "onload": [Function],
      "onresize": [Function],
      "onscroll": [Function],
      "onselect": [Function],
      "onsubmit": [Function],
      "ref": [Function],
    },
  },
  "html": "
<div class=\\"login-page\\">
  <form class=\\"form\\">
    <div class=\\"logo\\"><span><span>[</span> {{ logoText }} <span>]</span></span></div>
    <!--Logo-->
    <!--header-->
    <!--if-->
    <!--ErrorsPanel-->
    <div class=\\"form-group\\">
      <input name=\\"email\\" type=\\"text\\" placeholder=\\"Enter your email\\" autofocus=\\"\\">
    </div>
    <div class=\\"form-group\\">
      <input name=\\"password\\" type=\\"password\\" placeholder=\\"Enter your password\\">
    </div>
    <!--default-->
    <button type=\\"submit\\">Sign In
      <!--submit-->
    </button>
    <div class=\\"signup-label\\"> Not yet a member? <a href=\\"/\\"> Sign Up </a>
    </div>
    <!--footer-->
  </form>
</div>
",
}
`;
